frontend/apps/reader/modules/readersearch.lua

```diff
@@ -21,12 +37,26 @@ function ReaderSearch:init()
     self.ui.menu:registerToMainMenu(self)
 end
 
+local help_text = [[Here you can search text or a regular expression (ECMAScript syntax).
+
+Examples for regular expressions:
+If you search for all occurences of 'Mister Moore', 'Sir Moore' or 'Alfons Moore' but not for 'Lady Moore'.
+Enter 'Mister Moore|Sir Moore|Alfons Moore'.
+
+If you search 'run' and 'ran' you can enter 'r[ua]n'.
+
+If you want to search all numbers enter '[0-9]+'.
```

Comment:
Could be a bit less wordy, and a bit more an extended regexp reference:
```
'r[ua]n' will match 'run' and 'ran'
'[0-9]+' will match all numbers
```
I'd like to see something about how to anchor the match to word start and end.

I'll let the english introduction to @Frenzie :)

Reply:
Does word start/end refer to ^ and $ in this context?

Reply:
No, ^ and $ would match the start or end of a text node :)

http://www.akenotsuki.com/misc/srell/en/
There is 
\W | Equivalent to [^0-9A-Za-z_]. This can be used in a character class, too.
-- | --
\b | Out of a character class: matches a boundary between \w and \W. 				 				 				Inside a character class: matches BEL (\u0008).
-- | --
\B | Out of a character class: matches any boundary where \b does not match. 				 				  				Inside a character class: error_escape is thrown.


There is this (that we disabled) that could help getting something that works with non-ascii languages - but it would be really hard to document and use:
\p{...} | Matches any character that has the Unicode property specified in "...". For example, \p{scx=Latin} matches every character defined as a Latin letter in Unicode. This expression can be used also in a character class.
-- | --

Reply:
So you mean something like `[!\b]`? I don't know what you're asking me. ;-)

Reply:
I'm not asking _you_ specifically :)
Just thinking about the ways we can match words (to avoid matching other unrelated stuff).
It seems we only have `\b`: if we want to match `ion`, `ions`, `ionisation`, we can use `\bion` or `\bion\w+`, but that would work only for english/ascii. With french, there would be four word boundaries in ` tête `.

With keeping Unicode data, we could get better results with `\p{cat=!Letter|!Ideogram}ion` (whatever the right syntax is) - but go explain that :) So, it's fine if we don't have it.
Dunno if we should mention `\b` and it's limitation to ascii/english.

Reply:
I would not tell too much. I think most of the time alternatives (`James|Bond`) or single replacements (`r.n`) will be used.
To unleash the full potential the touch keyboards on our readers are not good enough.

Wouldn't it be a good article in our wiki: Sesame Street or explaining some use cases for regex.

For searching whole words starting with `mara` you can enter `(^| )mara[^ ]+`. (but you won't find `"Marathon"` this would need `(^| |")mara[^ ]+`)

Words containing an `uß` (which is non ascii) enter `[^ ]+uß[^ ]+`.

Reply:
> For searching whole words starting with mara you can enter ...

This won't match `(marathon de New-York 1996)`.

> I would not tell too much. I think most of the time alternatives 

Well, these alternatives deserves each one line :)
A wiki page won't be better than any other regex reference on the web.
I think we need a help_text with more examples, because the one time in the year a user will need this feature, he might be on the road :)

Reply:
> For searching whole words starting with mara 

`\Wmara` would work - so it needs being mentionned.
`uß\W` I guess too to match `schmuß` (dunno if that means something in german :)
But I guess `u\W` (finding words that end with `u`) would erroneously match `schmuß`.

Reply:
What do you think, if the regex help text is shown on hold of the regex checkbutton, not in the main menu?

Reply:
`Regular expression (hold for help)`

Reply:
> if the regex help text is shown on hold of the regex checkbutton, not in the main menu?

Yes, it's better: it avoids having to abandon what you entered till now to get back to the menu, and having to painfully enter the regex again :)

Reply:
Of course, you could add the opening `(` (but attention this has to be escaped) and you will end up with `(^| |"|\()mara[^ ]+`  -  only freaks can interpret. 
Personally I would go for a pragmatical solution with `mara[^ ]+` and ignore the false matches. If there are to many hits, I would improve the pattern (That's the way I do a google search.).
Maybe I would end up with `(^| ).?.?.?mara[^ ]+`

More examples - and your explanation for that - sounds reasonable. How long can/should the help_text be at maximum.

I would suggest we collect some simple examples in a kind of table:
Words containing `uß` -> `[^ ]+uß[^ ]+`
Alternative James, Bond, 007 -> `James|Bond|008`
Any single character -> `r.n`
Any characters ->  `J.*s`
Numbers -> `[0-9]+`

Reply:
It might be added to the end of the help text: "Hold the input field to copy string to the clipboard for further usage".

Reply:
> and having to painfully enter the regex again :)

The search text is cached, you don't have to enter it again :) as long the document is open.

Reply:
Can you check how `\W` works ?  It should be more practical than `(^| |"|\()`, one shouldn't have to mention all the possible punctuations/parens :)